packages/context/test/unit/resolver.test.ts

```diff
@@ -320,3 +326,90 @@ function customAsyncDecorator(def: Object) {
     return f + ':' + b;
   });
 }
+
+describe('method injection', () => {
+  let ctx: Context;
+
+  before(function() {
+    ctx = new Context();
+    ctx.bind('foo').to('FOO');
+    ctx.bind('bar').to('BAR');
+  });
+
+  it('can resolve method arguments', () => {
+    class TestClass {
+      test(@inject('foo') foo: string) {
+        return `hello ${foo}`;
+      }
+    }
+
+    const t = invokeMethod(TestClass.prototype, 'test', ctx);
+    expect(t).to.eql('hello FOO');
+  });
+
+  it('can resolve method arguments with instance', () => {
+    class TestClass {
+      bar: string;
+
+      test(@inject('foo') foo: string) {
+        this.bar = foo;
+        return `hello ${foo}`;
+      }
+    }
+
+    const inst = new TestClass();
+    const t = invokeMethod(inst, 'test', ctx);
+    expect(t).to.eql('hello FOO');
+    expect(inst.bar).to.eql('FOO');
+  });
+
+  it('can report error for missing binding key', () => {
+    class TestClass {
+      test(
+        @inject('', {x: 'bar'})
```

Comment:
Is there any particular reason why this is passing two arguments to `@inject`? I find it confusing. Can we use `@inject('key-does-not-exist')` instead?